### DISCUSSION TITLE
Pivot Points

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#274] [Indicators]
Community discussion about both the traditional [Pivot Points](https://dotnet.stockindicators.dev/indicators/PivotPoints/#content) and the modern [Rolling Pivot Points](https://dotnet.stockindicators.dev/indicators/RollingPivots/#content)

**Comment by DaveSkender** [comment]
The traditional calculations for the Pivot Points indicator uses fixed calendar windows to produce several unique flavors of support and resistance that look like flat bars.  I suspect it is done this way in part because it is an old indicator that pre-dates computers and 24-hour trading markets.

The static and somewhat arbitrary nature of the traditional calendar windows does not seem right to me — you’d rarely get level crosses at the beginning of a monthly window, simply because of the way it’s done and not because of market action, right?

To modernize it a bit, I’ve been thinking about doing a Rolling variant that would look more like BB or Donchian channels, but using the PP formulas for bands over a user-specified lookback period.  

Thoughts?

**Reply by DaveSkender** [reply]
This is what a monthly traditional Pivot Points (flat lines) versus a Rolling Pivot Point (squiggly lines) would look like:

![image](https://user-images.githubusercontent.com/8432125/103254572-cddf2a80-4953-11eb-8f0b-9b3e8ff5e3e3.png)

**Comment by GordonEldest** [comment]
Pivots Levels is so much used by traders that it acts as a self-fulfilling prophecy. (more that Fibo)

So, if you take a close look at your chart, you'll see that the stocks exhibit a certain magnetism toward either the gray line (PV) or the R1 when it is going up. (Nothing very significant in the left side when going down, this in itself is an indication of a certain hidden entropy)

A Rolling pivot doesn't add anything as it is not a used instrument of measure, and Pivots in itsefl is a weak concept.

What could be more interesting is to use the nearest Pivot line (in the Euclidian distance sense) to de-trend the low frequency of a curve. (Everybody seems to like to de-trend noise, but noise is interesting when it appears.)

**Reply by DaveSkender** [reply]
Yah, I agree with your assessment here.  I think the only reason I like rolling pivots over traditional pivot points is that is dispenses with this arbitrary fixed calendar window.  In the end though, I agree that rolling pivots don't add much to the overall concept.

As far as de-trending noise goes, I think there's better indicators in the moving average realm.  Rolling Pivots of [HTL](https://daveskender.github.io/Stock.Indicators/indicators/HtTrendline/#content) or [ALMA](https://daveskender.github.io/Stock.Indicators/indicators/Alma/#content) might be interesting to try as an [indicator of indicators](https://daveskender.github.io/Stock.Indicators/docs/GUIDE.html#generating-indicator-of-indicators).

**Reply by GordonEldest** [reply]
Yes I agree ALMA is probably the best ref for detrend, (with HMA.)  at least there is sounder math behind

Once that said back to Pivots:
I am dealing here with "Day" pivots. 
What I am pointing out is that noise analysis is interesting.  The "noise" level is an indication of an activity.  
(only when Bid is close to a Pivot level):
For thoses who doesn't have access to L2 or even L1 ticks, then candles "vibration" around a Pivots have some funny caracteristics 
and end up with a "make it" or "break it". 
Technics is to see this long trend around pivot and a PSAR for exemple can give a level to  decide Buy/Sale

Week, or longer durations, pivots are less interesting (IMHO). It is outperformed by "cup and handle" pattern analysis.

**Comment by Salardx** [comment]
Isn't there any other indicator that depic or estimate a trend line roughly? I mean these pivot lines are horizental support/resistance level,
what if we wanted to have sloping lines, acting as support/resistance?

**Reply by DaveSkender** [reply]
Pretty much any of the [price channels](https://daveskender.github.io/Stock.Indicators/indicators/#price-channels) will give you something like that.

**Comment by jeonnagroup2** [comment]
Hey dave why only the PP, R1,R2, S1 and S2 have data  ? i tried with the 1, 5, 15 minutes quotes and all the same. 

i compare it with the tradingview chart 

Maybe i missed something 
Btw fantastic library !! 

![image](https://user-images.githubusercontent.com/91985268/142011062-29bfa83e-90d8-4628-9e7f-40a43b7c8a9d.png)

![image](https://user-images.githubusercontent.com/91985268/142011228-7e074a65-4a06-4987-abe4-1adc6ae8f99d.png)

**Reply by DaveSkender** [reply]
There are several different flavors of Pivot Points.  By definition, some have more levels than other.

![image](https://user-images.githubusercontent.com/8432125/142043880-778a2862-f8a2-4cd5-83af-8ccbd90da0eb.png)

If it's the [Standard type](https://en.wikipedia.org/wiki/Pivot_point_(technical_analysis)) you're reviewing, it looks like we implemented the traditional 2 levels; though, I see that more levels can optionally be added.  I can take a look at adding those if you think they're useful?

- Related to #815

**Comment by jeonnagroup2** [comment]
nah 2 is cool for my needs.

**Comment by Gadirino** [comment]
hello @DaveSkender 
I wrote the pivot points code with python in vscode but vscode had some error in indicator parameter defining.
![p1](https://user-images.githubusercontent.com/110600525/183849884-f33228ee-3e66-486d-9fbb-599f604c335d.JPG)

What's my mistake?

**Reply by DaveSkender** [reply]
Tracking with https://github.com/DaveSkender/Stock.Indicators.Python/issues/244

**Comment by OthiCpp** [comment]
Hi Dave,
First, thank you very much for your library, do appreciate ! 

I want to use the  GetPivotPoints(windowSize, pointType) method, but the PeriodSize.FourHours is not expected by the method. 

Please if you can help !
Thank you.

**Reply by DaveSkender** [reply]
I'll have to think about this one.  Four weeks seems like a reasonable length to do this analysis; however, the algorithm is really dumb in that is simply boils down date to a component part for segmentation.  For example, `PointSize.Month` just simplifies the date to the month number.  For a four week window, we'd need to somehow determine which four weeks to use so it doesn't repaint new windows with every new bar that you add.

- Tracking with #1147

**Reply by OthiCpp** [reply]
Hi Dave, 
I am looking for 4 Hours not 4 Months. 
The 4 hours window raises the exception  : "Pivot Points does not support PeriodSize of FourHours"
Thank you in advance.

**Reply by DaveSkender** [reply]
Oh, sure.  Same issue.  Right now it's only setup to work with the traditional windows sizes, and will throw an exception on validation if you try to use a different `PeriodSize`.

![image](https://github.com/DaveSkender/Stock.Indicators/assets/8432125/480abd9d-22b6-42ed-8608-6f35d0444c35)

**Reply by OthiCpp** [reply]
Thank you for your reply. Hope will be added in the next release.

**Reply by DaveSkender** [reply]
Just to set some sad expectations, we’re focusing on the [next major v3 features](https://github.com/users/DaveSkender/projects/1/views/1) before we starting on any major indicator additions or updates.

With that said, we’re always looking for community support with [code contributions](https://dotnet.stockindicators.dev/contributing). 🥺
